src/Classes/Utils/CanvasUtils.ts

```diff
@@ -0,0 +1,348 @@
+        context.moveTo(begin.x, begin.y);
+        context.lineTo(end.x, end.y);
+        context.stroke();
+        context.closePath();
+    }
+
+    public static gradient(context: CanvasRenderingContext2D, p1: Particle, p2: Particle, midColor: string): CanvasGradient | undefined {
+        const gradStop = Math.floor(p2.radius / p1.radius);
+
+        if (!p1.color || !p2.color) {
+            return;
+        }
+
+        const sourcePos = p1.position;
+        const destPos = p2.position;
+        const grad = context.createLinearGradient(sourcePos.x, sourcePos.y, destPos.x, destPos.y);
+
+        grad.addColorStop(0, Utils.getStyleFromColor(p1.color));
+        grad.addColorStop(gradStop > 1 ? 1 : gradStop, midColor);
+        grad.addColorStop(1, Utils.getStyleFromColor(p2.color));
+
+        return grad;
+    }
+
+    public static drawGrabLine(context: CanvasRenderingContext2D, width: number, begin: ICoordinates, end: ICoordinates, colorLine: IRgb, opacity: number) {
+        context.strokeStyle = `rgba(${colorLine.r},${colorLine.g},${colorLine.b},${opacity})`;
+        context.lineWidth = width;
+        context.beginPath();
+        context.moveTo(begin.x, begin.y);
+        context.lineTo(end.x, end.y);
+        context.stroke();
+        context.closePath();
+    }
+
+    public static drawParticle(context: CanvasRenderingContext2D, particle: Particle, colorValue: string, backgroundMask: boolean, radius: number, stroke: IStroke) {
+        context.save();
+
+        // TODO: Performance issues, the canvas shadow is really slow
+        // const shadow = options.particles.shadow;
+
+        // if (shadow.enable) {
+        //     ctx.shadowBlur = shadow.blur;
+        //     ctx.shadowColor = shadow.color;
+        //     ctx.shadowOffsetX = shadow.offset.x;
+        //     ctx.shadowOffsetY = shadow.offset.y;
+        // } else {
+        //     delete ctx.shadowBlur;
+        //     delete ctx.shadowColor;
+        //     delete ctx.shadowOffsetX;
+        //     delete ctx.shadowOffsetY;
+        // }
+
+        context.fillStyle = colorValue;
+
+        const pos = {
+            x: particle.position.x,
+            y: particle.position.y,
+        };
+
+        context.translate(pos.x, pos.y);
+        context.beginPath();
+
+        if (particle.angle !== 0) {
+            context.rotate(particle.angle * Math.PI / 180);
+        }
+
+        if (backgroundMask) {
+            context.globalCompositeOperation = 'destination-out';
+        }
+
+        CanvasUtils.drawShape(context, particle, radius, stroke);
+
+        context.closePath();
+
+        if (stroke.width > 0) {
+            context.strokeStyle = stroke.color;
+            context.lineWidth = stroke.width;
+            context.stroke();
+        }
+
+        context.fill();
+        context.restore();
+    }
+
+    private static drawShape(context: CanvasRenderingContext2D, particle: Particle, radius: number, stroke: IStroke): void {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Exceeds maximum line length of 120](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5171185)